### PR TITLE
Redmine integration tests

### PR DESCRIPTION
**Pull request by SergeyMul**

### CODE REVIEW

File: test/webdriverio/specs/redmine.spec.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
+fdescribe("Redmine integration spec", function () {
+
+  var setupError;
+  var redmineUrl = 'http://demo.redmine.org';
+  var testProjectUrl = '';
+  var testIssueUrl = '';
+
+  var testProjectName = 'timetracker-redmine-test';
```
Comment (alexserdyuk) [inline test/webdriverio/specs/redmine.spec.js]: Don't use time tracker name on external services

File: test/webdriverio/customCommands.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
 var services = require('./services.conf');
 
+browser.addCommand("waitUrl", function (url, timeout) {
```
Comment (alexserdyuk) [inline test/webdriverio/customCommands.js]: Why do you need this command?

File: test/webdriverio/specs/redmine.spec.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+        href = result.value;
+      })
+      .click('.devart-timer-link')
+      .url('/')
+      .waitForExist('.portlet-body .timer-table')
+      .isVisible('#btn-stop')
+      .then(function (result) {
+        expect(result).toBeTruthy();
+      })
+      .getText('//tr[td/div/span[contains(.,"Active")]]/td[contains(@class,"timer-td-project")]/span').then(function (text) {
```
Comment (alexserdyuk) [inline test/webdriverio/specs/redmine.spec.js]: Why this code differs from jira.spec.js? We should re-use code.

File: test/webdriverio/specs/redmine.spec.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+      .waitForExist('.devart-timer-link')
+      .isExisting('.devart-timer-link-start')
+      .then(function (result) {
+        if (result) {
+          return browser
+            .click('.devart-timer-link-start')
+            .waitForExist('.devart-timer-link-stop');
+        }
+      })
+      .click('.devart-timer-link-stop')
```
Comment (alexserdyuk) [inline test/webdriverio/specs/redmine.spec.js]: Same question as above.

File: test/webdriverio/customCommands.js (hunk 2 above)
Comment (SergeyMul) [inline test/webdriverio/customCommands.js]: Page can be redirected to another url by clicking a submit button. During testing this situation can be handled by this command.

File: test/webdriverio/specs/redmine.spec.js (hunk 1 above)
Comment (SergeyMul) [inline test/webdriverio/specs/redmine.spec.js]: Ok

File: test/webdriverio/specs/redmine.spec.js (hunk 3 above)
Comment (SergeyMul) [inline test/webdriverio/specs/redmine.spec.js]: Factored code what test timetracker to special test commands in customCommand.js

File: test/webdriverio/specs/redmine.spec.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+    function createTestProject () {
+      return browser
+        .url(bugTrackerUrl + '/projects/new')
+        .setValue('#project_name', testProjectName)
+        .click('input[name=commit]')
+        .waitUrl({url: testProjectUrl + '/settings'})
+        .url(testProjectUrl)
+        .then(createTestIssue)
+        ;
```
Comment (ArtemAvramenko) [inline test/webdriverio/specs/redmine.spec.js]: Formatting

File: test/webdriverio/customCommands.js [hunk 6]
```diff
@@ -21,4 +30,102 @@ browser.addCommand("login", function (serviceName) {
         return new Error('Cannot login to ' + serviceName);
       }
     });
-}); 
\ No newline at end of file
+});
+
+browser.addCommand("testIsVisible", function (selector) {
```
Comment (alexserdyuk) [inline test/webdriverio/customCommands.js]: Commands testIsVisible, testIsNotVisible, testIsExisting, testIsNotExisting save only two lines of code, but make tests more obscure.

Lets use the the framework's code (sample below):

```js
browser
    .isVisible('#someRandomVisibleElement').then(function(isVisible) {
        expect(isVisible).toBeTruthy();
    })
```

**Pull request closed without merging**